File: app/src/main/java/com/rainbowguo/ethexplore/adapter/InternalTransactionAdapter.java
package com.rainbowguo.ethexplore.adapter;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.rainbowguo.ethexplore.MainActivity;
import com.rainbowguo.ethexplore.R;
import com.rainbowguo.ethexplore.Utils.TextUtils;
import com.rainbowguo.ethexplore.beans.internalTransactionsBean;
import com.rainbowguo.ethexplore.fragments.TransactionInfoFragment;

import java.util.ArrayList;

public class InternalTransactionAdapter extends RecyclerView.Adapter<InternalTransactionAdapter.mViewHolder> {
    private static final String TAG = "internalTransactionAdapter";
    private ArrayList<internalTransactionsBean.ResultDTO> internalTransactionList;
    private final String userAddress;

    public InternalTransactionAdapter(ArrayList<internalTransactionsBean.ResultDTO> list,String userAddress){
        this.internalTransactionList = list;
        this.userAddress = userAddress;
    }

    @NonNull
    @Override
    public mViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).
                inflate(R.layout.item_transaction,parent,false);
        Log.i(TAG, "onCreateViewHolder: " );
        return new mViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull mViewHolder holder, int position) {
        Log.i(TAG, "onBindViewHolder: binddata" );
        internalTransactionsBean.ResultDTO transaction = internalTransactionList.get(position);
        String fromAddress = transaction.getFrom();
        String toAddress = transaction.getTo();
        String time = transaction.getTimeStamp();
        String value = transaction.getValue();
        if (fromAddress.equals(userAddress)){
            holder.address.setText(toAddress);
            holder.role.setText("Send To");
        }
        else {
            holder.address.setText(fromAddress);
            holder.role.setText("Get from");
        }
        holder.time.setText(TextUtils.timeStampFormat(time));
        holder.value.setText(TextUtils.formatEther(value,null));
        holder.rootView.setOnClickListener(v->{
            MainActivity activity = (MainActivity) holder.itemView.getContext();
            TransactionInfoFragment fragment = new TransactionInfoFragment();
            Bundle bundle = new Bundle();
            bundle.putSerializable("data",transaction);
            fragment.setArguments(bundle);
            activity.addFragment(fragment);

            //holder.itemView.getContext().startActivity(new Intent(holder.itemView.getContext(), TransactionInfoActivity.class).putExtra("data",transaction));
        });
    }

    @Override
    public int getItemCount() {
        return internalTransactionList.size();
    }

    public static class mViewHolder extends RecyclerView.ViewHolder{
        public TextView role,address,time,value;
        public LinearLayout rootView;

        public mViewHolder(@NonNull View itemView) {
            super(itemView);
            role = itemView.findViewById(R.id.role);
            address = itemView.findViewById(R.id.address);
            time = itemView.findViewById(R.id.time);
            value = itemView.findViewById(R.id.value);
            rootView = itemView.findViewById(R.id.rootView);
        }
    }
}

File: app/src/main/java/com/rainbowguo/ethexplore/adapter/InternalTransactionAdapter.kt
package com.rainbowguo.ethexplore.adapter

import com.rainbowguo.ethexplore.beans.internalTransactionsBean
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import com.rainbowguo.ethexplore.R
import com.rainbowguo.ethexplore.adapter.InternalTransactionAdapter
import com.rainbowguo.ethexplore.MainActivity
import com.rainbowguo.ethexplore.fragments.TransactionInfoFragment
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.ImageView
import android.widget.TextView
import android.widget.LinearLayout
import com.rainbowguo.ethexplore.NewMainActivity
import com.rainbowguo.ethexplore.Utils.TextUtils
import java.util.ArrayList

class InternalTransactionAdapter(
    private val internalTransactionList: ArrayList<internalTransactionsBean.ResultDTO>,
    private val userAddress: String
) : RecyclerView.Adapter<InternalTransactionAdapter.mViewHolder>() {

    companion object {
        private const val TAG = "internalTransactionAdapter"
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): mViewHolder {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_transaction, parent, false)
        Log.i(TAG, "onCreateViewHolder: ")
        return mViewHolder(view)
    }

    override fun onBindViewHolder(holder: mViewHolder, position: Int) {
        Log.i(TAG, "onBindViewHolder: binddata")
        val transaction = internalTransactionList[position]
        val fromAddress = transaction.from
        val toAddress = transaction.to
        val time = transaction.timeStamp
        val value = transaction.value
        if (fromAddress == userAddress) {
            holder.address.text = toAddress
            holder.role.text = "Send To"
            holder.typeImg.setImageResource(R.drawable.get)
        } else {
            holder.address.text = fromAddress
            holder.role.text = "Get from"
            holder.typeImg.setImageResource(R.drawable.send)
        }
        holder.time.text = TextUtils.timeStampFormat(time)
        holder.value.text = TextUtils.formatEther(value, null)
        holder.rootView.setOnClickListener {
            val activity = holder.itemView.context as NewMainActivity
            val fragment = TransactionInfoFragment()
            val bundle = Bundle()
            bundle.putSerializable("data", transaction)
            fragment.arguments = bundle
            activity.addFragment(fragment)
        }
    }

    override fun getItemCount(): Int {
        return internalTransactionList.size
    }

    class mViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var role: TextView = itemView.findViewById(R.id.role)
        var address: TextView = itemView.findViewById(R.id.address)
        var time: TextView = itemView.findViewById(R.id.time)
        var value: TextView = itemView.findViewById(R.id.value)
        var rootView: LinearLayout = itemView.findViewById(R.id.rootView)
        var typeImg: ImageView = itemView.findViewById(R.id.type)

    }


}